font.go

```
@@ -222,3 +216,9 @@ func (f *Font) Underline() bool {
 func (f *Font) PointSize() int {
 	return f.pointSize
 }
+
+func screenDPI() int {
+	hDC := win.GetDC(0)
+	defer win.ReleaseDC(0, hDC)
+	return int(win.GetDeviceCaps(hDC, win.LOGPIXELSY))
```

It seems like before this set of calls happened once in walk init, but now it's made every time?

---

Screen DPI is dynamic now. On Windows 7 it was static - to change it, the user needed to logout/login - applications could safely consider it static in their UI. With Windows 8, it can change any time now when user changes 100%, 150%, 200% DPI...

We could cache it, but we do need to catch the windows message when the screen DPI is changed to update it. But, in the long run, we should eliminate or reduce calling `screenDPI()` as we UI should get the DPI from the current window it is painting on.

In my opinion, it should be optimized by reducing its usage.

---

Aren't we already redrawing on that message? In which case doesn't it make sense to grab the value then?

---

I have reviewed all the `screenDPI()` calls. They are needed in the depreciated functions for compatibility with the legacy clients (which would be better off defaulting to the hardcoded 96dpi value and stop declaring themselves as DPI-aware in the manifest for best appearance).

Some `screenDPI()` calls were in the methods on places where they should actually use the control/window DPI instead. Those were fixed in df794098bdd8444f2445d67f504995af6ea632e2.

---

The take-home message is: the `screenDPI()` function should be rarely called if ever. Its optimization is not essential.

---

FWIW, walk breaks compatibility all the time. Don't hesitate to break it here too, if it makes the code cleaner.